src/chttpd_db.erl

```diff
@@ -1127,12 +1155,20 @@ db_attachment_req(#httpd{method='GET',mochi_req=MochiReq}=Req, Db, DocId, FileNa
             % header we'll fall back to a chunked response.
             undefined
         end,
-        AttFun = case ReqAcceptsAttEnc of
+        AttFunTmp = case ReqAcceptsAttEnc of
         false ->
             fun couch_att:foldl_decode/3;
         true ->
             fun couch_att:foldl/3
         end,
+        case AttExternal of
+            "external" ->
+                AttFun = couch_att:fetch(data,Att),
+                couch_log:debug("chtttpd_db: Got new attachment ",[]);
+            _ ->
+                couch_log:debug("chttpd_db: not external att",[]),
+                AttFun = AttFunTmp
```

Comment:
So, AttFun could be actually an attachment data or some fold function? Something went wrong if you have to do this.